### PR TITLE
Change nematic example to conform to new API for freud 3.0

### PR DESCRIPTION
**Pull request by DomFijan**
## Description
Due to change in order.Nematic API the example requires changes. In addition many users were confused by the current example.  The new example will greatly expand on theory behind the nematic order parameter and introduce an easy way to compute a smectic order parameter as well alongside with some useful references for further reading. 

Resolves: #57  and glotzerlab/freud#945

## Checklist:

- [x] I agree with the terms of the [**freud Contributor Agreement**](https://github.com/glotzerlab/freud-examples/blob/master/ContributorAgreement.md).
- [ ] My code follows the code style of this project.
- [ ] I have updated the [index notebook](https://github.com/glotzerlab/freud-examples/blob/master/index.ipynb) if needed.
- [ ] My name is on the [contributors list](https://github.com/glotzerlab/freud-examples/blob/master/contributors.txt).

### CODE REVIEW

**Comment by tommy-waltmann** [comment]
I have gone through the start of the example and left some comments. I think this example could benefit from some more planning/storyboarding before doing a round of edits.

**Comment by DomFijan** [comment]
> I also think some changes in two files which don't have to do with the nematic example were accidentally committed.

pre-commit CI is to blame I think

**Comment by DomFijan** [comment]
@AlainKadar I made some initial edits according to the feedback. Can you take another look and see if text can be further improved? 

I wouldn't want to completely remove the mathematical details of how this is done since this is super hard to find online and I think it is valuable to have it here. Also, the implementation in freud follows the formulas laid out in the notebook anyways. What are your thoughts on this?

File: module_intros/density.LocalDensity.ipynb [hunk 1]
```diff
@@ -31,6 +31,8 @@
    "outputs": [],
```
Comment (tommy-waltmann) [inline module_intros/density.LocalDensity.ipynb:31]: <p>Let's remove the changes to this file.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/freud-examples/pull/68/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

File: module_intros/density.RDF-BinWidth.ipynb [hunk 2]
```diff
@@ -27,6 +27,8 @@
    "outputs": [],
```
Comment (tommy-waltmann) [inline module_intros/density.RDF-BinWidth.ipynb:27]: <p>Let's remove the changes to this file.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/freud-examples/pull/68/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

**Comment by DomFijan** [comment]
<p>We don't normalize it explicitly anywhere. </p>

---

 <a href='https://app.reviewnb.com/glotzerlab/freud-examples/pull/68/discussion/#comment-1733951140'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by tommy-waltmann** [comment]
@DomFijan I have gone through and re-worded the examples in this PR. Have another look over it and let me know if you have any further edits.

**Comment by DomFijan** [comment]
@tommy-waltmann 
This is good. The tests are failing due to changes in gsd.open read/write modes in some other notebooks.

**Review by tommy-waltmann — APPROVED** [review]
LGTM

**Pull request merged**